src/components/menu/menu.test.js

```diff
@@ -0,0 +1,1260 @@
+
+        cy.viewport(1200, 800);
+        getComponent("menu-fullscreen").should(state);
+      }
+    );
+
+    it.each([
+      ["left", -1200, 1200, 0],
+      ["right", 1200, -1200, 0],
+    ])(
+      "should verify that Menu Fullscreen start position is %s",
+      (side, left, right, after) => {
+        CypressMountWithProviders(
+          <MenuComponentFullScreen startPosition={side} />
+        );
+
+        cy.viewport(1200, 800);
+        getComponent("menu-fullscreen").should("have.css", "left", `${left}px`);
+        getComponent("menu-fullscreen").should(
+          "have.css",
+          "right",
+          `${right}px`
+        );
+        menuItem().eq(positionOfElement("first"), div).click();
+        getComponent("menu-fullscreen").should(
+          "have.css",
+          "left",
+          `${after}px`
+        );
+        getComponent("menu-fullscreen").should(
+          "have.css",
+          "right",
+          `${after}px`
+        );
+      }
+    );
+
+    it("should verify that the width of the whole submenu with a long-labelled item is determined by the width of the item", () => {
+      CypressMountWithProviders(
+        <Box mb={150}>
+          {["white"].map((menuType) => (
+            <div key={menuType}>
+              <Menu menuType={menuType}>
+                <MenuItem submenu="Menu Item One">
+                  <MenuItem href="#">
+                    Item Submenu One Is A Very Long Submenu Item Indeed
+                  </MenuItem>
+                  <MenuItem variant="alternate" href="#">
+                    Item Submenu Two
+                  </MenuItem>
+                </MenuItem>
+              </Menu>
+            </div>
+          ))}
+        </Box>
+      );
+
+      submenu().eq(positionOfElement("first")).trigger("mouseover");
+      innerMenu(positionOfElement("second"), span).then(($item) => {
+        submenuBlock().should("have.css", "width", `${$item.width()}px`);
+      });
+    });
+
+    it("should verify that the width of the whole submenu with a long title is determined by the width of the title", () => {
+      CypressMountWithProviders(
+        <Box mb={150}>
+          {["white"].map((menuType) => (
+            <div key={menuType}>
+              <Menu menuType={menuType}>
+                <MenuItem submenu="Menu Item One Has A Very Long Menu Title For No Reason Whatsoever">
+                  <MenuItem href="#">Item Submenu One</MenuItem>
+                  <MenuItem variant="alternate" href="#">
+                    Item Submenu Two
+                  </MenuItem>
+                </MenuItem>
+              </Menu>
+            </div>
+          ))}
+        </Box>
+      );
+
+      submenu().eq(positionOfElement("first")).trigger("mouseover");
+      submenu()
+        .eq(positionOfElement("first"))
+        .then(($menu) => {
+          submenuBlock().should("have.css", "width", `${$menu.width()}px`);
+        });
+    });
+
+    it.each([
```

Comment:
we can wrap this test into a separate `describe` and pass what exactly `type` we are passing, as I did in line 104 in `pages.test.js` file